cax/tasks/rucio_mover.py

```diff
@@ -0,0 +1,1557 @@
+      """Gather meta tag information"""
+      
+      #data base entry according the run
+      tags_all = self.run_doc
+      meta_tags = []
+
+      #Gather basic meta data:
+      meta_tag_shifter      = tags_all["user"]
+      meta_tag_name         = tags_all["name"]              #just an option, not used in meta data set (see w)
+      meta_tag_source_type  = tags_all["source"]["type"]
+      meta_tag_runnumber    = tags_all["number"]
+      meta_tag_sub_detector = tags_all["detector"]
+      meta_tag_trigger_events_built = tags_all["trigger"]["events_built"]
+      tag_created_at        = tags_all["comments"][0]["date"]
+      meta_tag_created_at   = tag_created_at.date().strftime("%Y%m%d")[2:]+"_"+tag_created_at.time().strftime("%H%M")
+      
+      #define the muon veto short term 'mv' (hardcoded)
+      if meta_tag_sub_detector == "muon_veto":
+        meta_tag_sub_detector = "mv"
+      
+      w = { "phys_group": meta_tag_shifter,
+            "provenance": meta_tag_sub_detector,
+            "datatype" : meta_tag_source_type,
+            "campaign"  : ("SR"+config.RUCIO_CAMPAIGN),
+            "run_number": meta_tag_runnumber
+            }
+      
+      meta_tags.append(w)
+      
+      #Add location based meta data:
+      if self.is_transferred_to_location(location, ttype) == True and ttype == "raw":
+        meta_tags[0].update( {"prod_step": "raw"} )
+      elif self.is_transferred_to_location(location, ttype) == True and ttype == "processed":
+        meta_tags[0].update( {"prod_step": "processed"} )
+      else:
+        meta_tags[0].update( {"prod_step": "undefined"} )
+      
+      meta_tags[0].update( {"version" : self.get_software_version( location, ttype)} )
+
+      return meta_tags
+    
+    #def create_script(self, scriptname, code):
+        #"""Create script as temp file to be run on cluster"""
+        
+        #f = open( scriptname, "w")
+        #f.write(code)
+        #f.close()
+        
+        #os.chmod(scriptname, 0o774)
+
+    def create_script(self, script):
+        """Create script as temp file to be run on cluster"""
+        fileobj = tempfile.NamedTemporaryFile(delete=False,
+                                            suffix='.sh',
+                                            mode='wt',
+                                            buffering=1)
+        fileobj.write(script)
+        os.chmod(fileobj.name, 0o774)
+
+        return fileobj
+
+    def delete_script(self, scriptname):
+        """Delete script after submitting to cluster
+
+        :param script_path: path to the script to be removed
+
+        """
+        subprocess.Popen(['rm', '-f', scriptname], stdout=subprocess.PIPE)
+
+    def get_python_version(self):
+      """Check for the right python version. Assure that rucio is executed under python <=2.7"""
+      sc = "/tmp/{sc_name}.sh".format(sc_name="python_check")
+      upload_string = """#.bashrc
+export PATH=/home/SHARED/anaconda3/bin:$PATH
```

Comment:
I just realize that I do not use this member function at all. But yeah -> Needs to be generalized.